Source/Provers/SMTLib/SMTLibProcess.cs

```diff
@@ -92,12 +92,18 @@ void ControlCHandler(object o, ConsoleCancelEventArgs a)
       }
     }
 
+    public override void IndicateEndOfInput()
+    {
+      prover.StandardInput.Close();
+      toProver = null;
```

Comment:
Instead of setting this to null here and checking for null below, could we check whether toProver has been closed?

Reply:
It doesn't look like there's a way to check that. :(